curriculum/challenges/english/14-responsive-web-design-22/learn-html-by-building-a-cat-photo-app/62bb4009e3458a128ff57d5d.md

```diff
@@ -7,14 +7,19 @@ dashedName: step-68
 
 # --description--
 
-One more thing. You should allow people to use their native language. Tell the browser to encode multiple languages by adding a `meta` element as a child of the `head` element. Set its `charset` attribute to `UTF-8`.
+You can set the browser behavior by using `meta` element. A famous use case is telling the browser to parse the markdown into multiple languages by setting `charset` attribute to `UTF-8`.
+
+Tell the browser to parse the markdown into multiple languages by creating a `meta` element as a child of the `head` element. Set its `charset` attribute to `UTF-8`.
 
 # --hints--
 
-You should create a `meta` element within the `head` element.
+You should create a self-closing `meta` element within the `head` element.
 
 ```js
 assert.exists(document.querySelector('head > meta'));
+let innerText = document.querySelector(" head ").innerHTML;
+let matched = (innerText.match(/meta/g) || []).length
+assert.equal(matched, 1)
```

Comment:
Hi @hanswang123456, the current test doesn't stop `<meta></meta>`

If you need help solving this, join us in [our chat room](https://discord.gg/PRyKn3Vbay) or [the forum](https://forum.freecodecamp.org/c/contributors/3)

Reply:
Sounds good, Ill join the chatroom

Reply:
@hanswang123456 are you still stuck?

Reply:
Sorry for the late response, a challenge I've been running into is detecting the closing meta tag's existence. When users include the tag, it is ignored by the browser. I'll need to use another method other than innerHTML(which does not return the closing meta tag)

Reply:
No worries - keep up the excellent work.

Reply:
From other review thread:

> I don't think the HTML specs allow for a closing </meta> tag and the W3 HTML validator throws an error if you have one. My understanding is that we want to allow valid HTML wherever possible. Conversely, I think we should fail invalid HTML as well.

While the validator may fail with this, the browser still renders the original meta tag as expected and just ignores the closing tag. Typically, I might say that we don't need to test things like this because the browser gives us what we expect and we can't test every edge case. But this is the first project, I would be fine with a test for it. 

I would add a third test for it @hanswang123456, the first test can be reverted to what it was (just check that the meta tag exists). The second one can stay as it is. And the third test can be `You should not have a closing meta tag` or `You should not have a closing </meta> tag`. The test will have to just use regex since nothing is rendered if a closing tag is added.

Reply:
> From other review thread:
> 
> > I don't think the HTML specs allow for a closing  tag and the W3 HTML validator throws an error if you have one. My understanding is that we want to allow valid HTML wherever possible. Conversely, I think we should fail invalid HTML as well.
> 
> While the validator may fail with this, the browser still renders the original meta tag as expected and just ignores the closing tag. Typically, I might say that we don't need to test things like this because the browser gives us what we expect and we can't test every edge case. But this is the first project, I would be fine with a test for it.
> 
> I would add a third test for it @hanswang123456, the first test can be reverted to what it was (just check that the meta tag exists). The second one can stay as it is. And the third test can be `You should not have a closing meta tag` or `You should not have a closing </meta> tag`. The test will have to just use regex since nothing is rendered if a closing tag is added.

Sounds good. I'll update the 3rd test .

Reply:
Thank you @hanswang123456, I have update the test to close the issue. Thank you for the hard work and bearing with us